bouncer/shared/contract_swap.ts

```diff
@@ -87,16 +87,50 @@ export async function performSwapViaContract(
 
   const tag = swapTag ?? '';
 
+  const srcChain = chainFromAsset(sourceAsset);
+
+  // Generate a new wallet for each contract swap to prevent nonce issues when running in parallel
+  // with other swaps via deposit channels.
+  const mnemonic = Wallet.createRandom().mnemonic?.phrase ?? '';
+  if (mnemonic === '') {
+    throw new Error('Failed to create random mnemonic');
+  }
+  const wallet = Wallet.fromPhrase(mnemonic).connect(getDefaultProvider(getEvmEndpoint(srcChain)));
+
   try {
+    // Fund new key with native asset and asset to swap.
+    await send(chainGasAsset(srcChain), wallet.address);
+    await send(sourceAsset, wallet.address);
+
+    if (erc20Assets.includes(sourceAsset)) {
+      // Doing effectively infinite approvals to make sure it doesn't fail.
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      await approveTokenVault(
+        sourceAsset,
+        (
+          BigInt(amountToFineAmount(defaultAssetAmounts(sourceAsset), assetDecimals(sourceAsset))) *
```

Comment:
we should pull this out into a function that takes a source asset and amount. We use this pattern of `BigInt(amountToFineAmount...` in a number of other places too